### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by nhokpro** [httptoolkit/httptoolkit#358]
### Repro steps

The mock function does not fully respond to all headers with the same name
I want to mock a request to my host with content 2 set-cookie headers.
The function auto replaces the first header with the second value
It should have 2 headers
set-cookie a=b
set-cookie c=d
This applied to all headers with the same name as well
 

### How often does this bug happen?

Every time

### The desktop OS you're using

windows 10

**Comment by pimterry** [comment]
Can you explain exactly what steps you're doing when you see this? What type of rule are you using, what values exactly have you entered, and where do you see that only one set-cookie is shown? A list of steps so I can reproduce exactly what you're seeing would be very helpful.

I've just tested myself, and for the fixed-response case this seems to work fine:

![Screenshot from 2022-11-22 13-39-04](https://user-images.githubusercontent.com/1526883/203316295-0d7a001b-ea15-4b3b-81c7-20f70d2062e2.png)

results in this response from Chrome's POV:

![Screenshot from 2022-11-22 13-39-17](https://user-images.githubusercontent.com/1526883/203316306-c5ec436c-aa5f-488f-8588-ec53be623268.png)

**Comment by nhokpro** [comment]
I mean pass the request to my host
My host will return 2 headers set-cookie
But the httptoolkit only get 1 and return back to the response header only 1

**Comment by pimterry** [comment]
It's not clear what "mock the request to localhost" means, there are a lot of different ways to mock requests with HTTP Toolkit and a lot of different ways that the client & server might be behaving depending on exactly what you're doing.

Can you please provide specific and complete steps to reproduce this issue? There are a lot of details here that aren't clear. For example:

* Open HTTP Toolkit
* Create a rule that says... [screenshot]
* Start an intercepted Chrome window
* Open URL X (a public URL I can test)
* HTTP Toolkit shows Y [screenshot]
* Chrome shows Z [screenshot]

**Comment by nhokpro** [comment]
![image](https://user-images.githubusercontent.com/31965576/203391740-82823bef-3711-400a-949e-2fa677b041e0.png)

**Comment by pimterry** [comment]
Sorry, but I still cannot reproduce this - there's something that's missing here.

When I open m.facebook.com in an intercepted Chrome window, using HTTP Toolkit with no rules and default settings, everything appears correctly and I receive two set-cookie headers.

Intercepted headers in HTTP Toolkit:

![Screenshot from 2022-11-23 18-02-05](https://user-images.githubusercontent.com/1526883/203606365-5710d231-c0d5-4030-a7ef-800a935c6b5c.png)

Resulting response in Chrome:
![Screenshot from 2022-11-23 18-02-17](https://user-images.githubusercontent.com/1526883/203606371-bfd2ad38-c891-44cc-994d-80f74c22106e.png)

If you can work out what triggers this, please let me know and I'll investigate further.

**Comment by nhokpro** [comment]
> Sorry, but I still cannot reproduce this - there's something that's missing here.
> 
> When I open m.facebook.com in an intercepted Chrome window, using HTTP Toolkit with no rules and default settings, everything appears correctly and I receive two set-cookie headers.
> 
> Intercepted headers in HTTP Toolkit:
> 
> ![Screenshot from 2022-11-23 18-02-05](https://user-images.githubusercontent.com/1526883/203606365-5710d231-c0d5-4030-a7ef-800a935c6b5c.png)
> 
> Resulting response in Chrome: ![Screenshot from 2022-11-23 18-02-17](https://user-images.githubusercontent.com/1526883/203606371-bfd2ad38-c891-44cc-994d-80f74c22106e.png)
> 
> If you can work out what triggers this, please let me know and I'll investigate further.

![image](https://user-images.githubusercontent.com/31965576/203641698-ce03f07a-5e6b-433a-b8dd-77ef9c91db30.png)
I am not quite sure, but something went wrong with the HTTP toolkit header handle
Can you login into Facebook with your account when Intercepting?
Normal chrome will set the c_user header and xs header.
My phone notify me that I have access to facebook on the new device, but in the Intercepted chrome, it will not set the cookies, so the site will act like not login.

**Comment by pimterry** [comment]
> Can you login into Facebook with your account when Intercepting?

Yes, everything works fine - multiple cookies are set with no problem and I can log in and use the Facebook like normal.

Can you try installing HTTP Toolkit on a different machine, and test that it works there? I would be surprised if it doesn't, but it's good to check and it's always possible that this is a network issue somehow.

If that does work, then it must be something to do with the configuration on your computer somewhere - either an HTTP Toolkit rule or option on the Settings page, or something to do with your computer's settings more generally.

If you want, you can reset HTTP Toolkit's settings completely to check this. To do so, open the developer tools (View -> Toggle developer tools), then go to Application -> Storage, click 'Clear site data' (leaving all checkboxes below ticked), and then close and reopen the app. Be aware this will log you out of Pro, so you'll have to send yourself a new code.
